app/controllers/tmdb_controller.rb

```diff
@@ -134,53 +137,12 @@ def director_search
   end
 
   def discover_search
-    #format date/year hash passed in params. otherwise year is passed directly on paginated pages
-    params[:year] = params[:date][:year] if params[:date].present?
+    form_params = %i[sort_by date genre actor actor2 company mpaa_rating year_select page]
+    passed_params = params.slice(*form_params).select { |_k, v| v.present? }
+    return if passed_params.blank?
 
-    @passed_params = params.slice(:sort_by, :year, :genre, :actor, :actor2,
-      :company, :mpaa_rating, :year_select, :page).select{ |k, v| v.present?}
-
-    if @passed_params.any?
-      parse_params(@passed_params) #module to help parse
-
-      #parse passed params to show user what they searched for
-      @discover_view_params = @passed_params.slice(:actor, :genre, :date, :year, :year_select, :mpaa_rating, :sort_by)
-      @params_for_view = discover_show_search_params(@discover_view_params)
-
-      tmdb_handler_discover_search(@passed_params)
-    end
-
-  end #discover search
-
-  def discover_show_search_params(show_params)
-    @keys = show_params.keys
-    @actor_display = show_params[:actor].titlecase if @keys.include?("actor")
-
-    if @keys.include?("genre")
-      @genre_id = show_params[:genre].to_i
-      @genres = Movie::GENRES.to_h
-      @genre_selected = @genres.key(@genre_id)
-      @genre_display = "#{@genre_selected} movies"
-    end
-
-    @rating_display = "Rating: #{show_params[:mpaa_rating]}" if @keys.include?("mpaa_rating")
-
-    if show_params[:year_select] == "exact" || !show_params[:year_select].present?
-      @year_select_display = "From"
-    else
-      @year_select_display = show_params[:year_select]
-    end
-
-    @year_show = show_params[:year] if @keys.include?("year")
-    @year_display = "#{@year_select_display} #{@year_show}" if @year_show.present?
-
-    if @keys.include?("sort_by")
-      @sort_selected = show_params[:sort_by]
-      @sort_options = Movie::SORT_BY.to_h
-      @sort_key = @sort_options.key(@sort_selected)
-      @sort_display = "sorted by #{@sort_key}" if @keys.include?("sort_by")
-    end
-    "#{@actor_display} #{@genre_display} #{@rating_display} #{@year_display} #{@sort_display}"
+    searchable_params = SearchParamParser.parse_movie_params(passed_params)
+    tmdb_handler_discover_search(searchable_params)
+    @params_for_view = SearchParamParser.parse_movie_params_for_display(passed_params)
```

Comment:
How do you feel about some other variable name?

Since the method returns a string, it might be clearer to name it something like `@parsed_results_text` so that it's not confused for more params?

Reply:
`@params_for_view` is what is currently in-use in the view. I was trying to not touch the instance variables in the view for this PR. I'll totes change it in the next one where I mess around in the view though.

Reply:
👍 Good plan!